libbpf-tools/sigsnoop.c

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+
+/*
+ * sigsnoop	Trace standard and real-time signals.
+ *
+ * Copyright (c) 2021~2022 Hengqi Chen
+ *
+ * 08-Aug-2021   Hengqi Chen   Created this.
+ */
+#include <argp.h>
+#include <libgen.h>
+#include <signal.h>
+#include <time.h>
+
+#include <bpf/bpf.h>
+#include "sigsnoop.h"
+#include "sigsnoop.skel.h"
+
+#define PERF_BUFFER_PAGES	16
+#define PERF_POLL_TIMEOUT_MS	100
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+
+static volatile sig_atomic_t exiting = 0;
+
+static pid_t target_pid = 0;
+static int target_signal = 0;
+static bool failed_only = false;
+static bool kill_only = false;
+static bool signal_name = false;
+static bool verbose = false;
+
+static const char *sig_name[] = {
+	[0] = "N/A",
+	[1] = "SIGHUP",
```

Comment:
What do you think of using signal macro instead of its value, like `[SIGHUP] = "SIGHUP"`?

You can apply this patch to translate all of them:

```patch
diff --git a/libbpf-tools/sigsnoop.c b/libbpf-tools/sigsnoop.c
index 279139b0..b8e98cfa 100644
--- a/libbpf-tools/sigsnoop.c
+++ b/libbpf-tools/sigsnoop.c
@@ -31,38 +31,38 @@ static bool verbose = false;
 
 static const char *sig_name[] = {
        [0] = "N/A",
-       [1] = "SIGHUP",
-       [2] = "SIGINT",
-       [3] = "SIGQUIT",
-       [4] = "SIGILL",
-       [5] = "SIGTRAP",
-       [6] = "SIGABRT",
-       [6] = "SIGIOT",
-       [7] = "SIGBUS",
-       [8] = "SIGFPE",
-       [9] = "SIGKILL",
-       [10] = "SIGUSR1",
-       [11] = "SIGSEGV",
-       [12] = "SIGUSR2",
-       [13] = "SIGPIPE",
-       [14] = "SIGALRM",
-       [15] = "SIGTERM",
-       [16] = "SIGSTKFLT",
-       [17] = "SIGCHLD",
-       [18] = "SIGCONT",
-       [19] = "SIGSTOP",
-       [20] = "SIGTSTP",
-       [21] = "SIGTTIN",
-       [22] = "SIGTTOU",
-       [23] = "SIGURG",
-       [24] = "SIGXCPU",
-       [25] = "SIGXFSZ",
-       [26] = "SIGVTALRM",
-       [27] = "SIGPROF",
-       [28] = "SIGWINCH",
-       [29] = "SIGIO",
-       [30] = "SIGPWR",
-       [31] = "SIGSYS",
+       [SIGHUP] = "SIGHUP",
+       [SIGINT] = "SIGINT",
+       [SIGQUIT] = "SIGQUIT",
+       [SIGILL] = "SIGILL",
+       [SIGTRAP] = "SIGTRAP",
+       [SIGABRT] = "SIGABRT",
+       [SIGIOT] = "SIGIOT",
+       [SIGBUS] = "SIGBUS",
+       [SIGFPE] = "SIGFPE",
+       [SIGKILL] = "SIGKILL",
+       [SIGUSR1] = "SIGUSR1",
+       [SIGSEGV] = "SIGSEGV",
+       [SIGUSR2] = "SIGUSR2",
+       [SIGPIPE] = "SIGPIPE",
+       [SIGALRM] = "SIGALRM",
+       [SIGTERM] = "SIGTERM",
+       [SIGSTKFLT] = "SIGSTKFLT",
+       [SIGCHLD] = "SIGCHLD",
+       [SIGCONT] = "SIGCONT",
+       [SIGSTOP] = "SIGSTOP",
+       [SIGTSTP] = "SIGTSTP",
+       [SIGTTIN] = "SIGTTIN",
+       [SIGTTOU] = "SIGTTOU",
+       [SIGURG] = "SIGURG",
+       [SIGXCPU] = "SIGXCPU",
+       [SIGXFSZ] = "SIGXFSZ",
+       [SIGVTALRM] = "SIGVTALRM",
+       [SIGPROF] = "SIGPROF",
+       [SIGWINCH] = "SIGWINCH",
+       [SIGIO] = "SIGIO",
+       [SIGPWR] = "SIGPWR",
+       [SIGSYS] = "SIGSYS",
 };
 
 const char *argp_program_version = "sigsnoop 0.1";
 ```

Reply:
I am inclined to keep as it is. Because by reading the code I know which number corresponds to which signal.

Reply:
Understandable since your tool can print both the signal as integer value and its string name.
So, by looking at the code you get the translation.